src/Tools/dotnet-trace/CommandLine/Commands/ListCommandHandler.cs

```diff
@@ -0,0 +1,38 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using Microsoft.Diagnostics.Tracing.Eventing;
+using System;
+using System.CommandLine;
+using System.CommandLine.Invocation;
+using System.Threading.Tasks;
+
+namespace Microsoft.Diagnostics.Tools.Trace
+{
+    internal static class ListCommandHandler
+    {
+        public static async Task<int> AvailablePorts(IConsole console)
+        {
+            try
+            {
+                foreach (var pid in EventPipeClient.ListAvailablePorts())
+                    Console.Out.WriteLine($"ProcessId={pid}");
+
+                await Task.FromResult(0);
+                return 0;
+            }
+            catch (Exception ex)
+            {
+                Console.Error.WriteLine($"[ERROR]: {ex.ToString()}");
+                return 1;
+            }
+        }
+
+        public static Command ListCommand() =>
```

Comment:
You probably need to add the "list" subcommand to the spec.

Reply:
@jorive mentioned to this to me offline that he wanted it as undocumented helper function.

Thinking on it a little more I got worried that at some point in the future we will want a list command that lists active tracing sessions rather than listing runtime ports and this command will be lurking in our way. @jorive if you are very confident you will have time to take this back out before Preview5 then I'm fine leaving it as-is. If you have any doubts that maybe it will get forgotten or time crunch will come and you won't get back to it then how about as an alternative we create a dotnet-ps tool. We don't have to ship it, but you can put this functionality in it and use it whenever you want internally.

Reply:
I've been thinking about it to. I might remove it before merge, and have a python tool for my purposes.